main/main.go

```diff
@@ -0,0 +1,298 @@
+			Name:  "input-type",
+			Usage: "currently json and yaml are supported. If not set, sops will use the file's extension to determine the type",
+		},
+		cli.StringFlag{
+			Name:  "output-type",
+			Usage: "currently json and yaml are supported. If not set, sops will use the input file's extension to determine the output format",
+		},
+		cli.BoolFlag{
+			Name:  "show-master-keys, s",
+			Usage: "display master encryption keys in the file during editing",
+		},
+		cli.StringFlag{
+			Name:  "add-kms",
+			Usage: "add the provided comma-separated list of KMS ARNs to the list of master keys on the given file",
+		},
+		cli.StringFlag{
+			Name:  "rm-kms",
+			Usage: "remove the provided comma-separated list of KMS ARNs from the list of master keys on the given file",
+		},
+		cli.StringFlag{
+			Name:  "add-pgp",
+			Usage: "add the provided comma-separated list of PGP fingerprints to the list of master keys on the given file",
+		},
+		cli.StringFlag{
+			Name:  "rm-pgp",
+			Usage: "remove the provided comma-separated list of PGP fingerprints from the list of master keys on the given file",
+		},
+		cli.BoolFlag{
+			Name:  "ignore-mac",
+			Usage: "ignore Message Authentication Code during decryption",
+		},
+		cli.StringFlag{
+			Name:  "unencrypted-suffix",
+			Usage: "override the unencrypted key suffix. default: unencrypted_",
+			Value: sops.DefaultUnencryptedSuffix,
+		},
+		cli.StringFlag{
+			Name:  "config",
+			Usage: "path to sops' config file. If set, sops will not search for the config file recursively.",
+		},
+	}
+	app.Action = func(c *cli.Context) error {
+		if c.NArg() < 1 {
+			return cli.NewExitError("Error: no file specified", 1)
+		}
+		file := c.Args()[0]
+		if _, err := os.Stat(file); os.IsNotExist(err) {
+			if c.String("add-kms") != "" || c.String("add-pgp") != "" || c.String("rm-kms") != "" || c.String("rm-pgp") != "" {
+				return cli.NewExitError("Error: cannot add or remove keys on non-existent files, use `--kms` and `--pgp` instead.", 49)
+			}
+			if c.Bool("encrypt") || c.Bool("decrypt") || c.Bool("rotate") {
+				return cli.NewExitError("Error: cannot operate on non-existent file", 100)
+			}
+		}
+		fileBytes, err := ioutil.ReadFile(file)
+
+		var output *os.File
+		if c.Bool("in-place") {
+			var err error
+			output, err = os.Create(file)
+			if err != nil {
+				return cli.NewExitError(fmt.Sprintf("Could not open in-place file for writing: %s", err), 10)
+			}
+			defer output.Close()
+		} else {
+			output = os.Stdout
+		}
+		if err != nil {
+			return cli.NewExitError("Error: could not read file", 2)
+		}
+		if c.Bool("encrypt") {
+			return encrypt(c, file, fileBytes, output)
+		} else if c.Bool("decrypt") {
+			return decrypt(c, file, fileBytes, output)
+		} else if c.Bool("rotate") {
+			return rotate(c, file, fileBytes, output)
+		} else {
+
+		}
+		return nil
+	}
+	app.Run(os.Args)
+}
+
+func runEditor(path string) {
+	editor := os.Getenv("EDITOR")
+	var cmd *exec.Cmd
+	if editor == "" {
```

Comment:
You should inverse the logic here: `if os.Getenv("EDITOR") != "" { prepare cmd with that value} else { prepare cmd with vim or nano }`